keystorage/aws/src/test/java/tech/pegasys/signers/aws/AwsSecretsManagerTest.java

```diff
@@ -73,255 +73,208 @@ private void verifyEnvironmentVariables() {
     assertThat(secret).hasValue(SECRET_VALUE);
   }
 
   @Test
   void fetchSecretWithExplicitManager() {
-    createSecret(false, false, false);
-    Optional<String> secret = awsSecretsManagerExplicit.fetchSecret(secretName);
+    Optional<String> secret = awsSecretsManagerExplicit.fetchSecret(secretName1);
     assertThat(secret).hasValue(SECRET_VALUE);
   }
 
   @Test
   void fetchSecretWithInvalidCredentialsReturnsEmpty() {
-    createSecret(false, false, false);
     assertThatExceptionOfType(RuntimeException.class)
-        .isThrownBy(() -> awsSecretsManagerInvalidCredentials.fetchSecret(secretName))
+        .isThrownBy(() -> awsSecretsManagerInvalidCredentials.fetchSecret(secretName1))
         .withMessageContaining("Failed to fetch secret from AWS Secrets Manager.");
   }
 
   @Test
   void fetchingNonExistentSecretReturnsEmpty() {
-    createSecret(false, false, false);
     Optional<String> secret = awsSecretsManagerDefault.fetchSecret("signers-aws-integration/empty");
     assertThat(secret).isEmpty();
   }
 
+  // emptyTagFiltersReturnAllKeys: search([], []) returns all secrets
   @Test
-  void listAndMapSingleSecretWithSingleTag() {
-    createSecret(false, false, false);
-
+  void emptyTagFiltersReturnAllSecrets() {
     final Collection<AbstractMap.SimpleEntry<String, String>> secretEntries =
         awsSecretsManagerExplicit.mapSecrets(
-            secretTags.keySet().stream().collect(Collectors.toList()),
-            secretTags.values().stream().collect(Collectors.toList()),
-            AbstractMap.SimpleEntry::new);
+            Collections.emptyList(), Collections.emptyList(), AbstractMap.SimpleEntry::new);
 
-    secretNames.forEach(secretName -> validateMappedSecret(secretEntries, secretName));
+    final List<String> secretNames =
+        secretEntries.stream()
+            .map(entry -> String.valueOf(entry.getKey()))
+            .collect(Collectors.toList());
+    assertThat(secretNames).contains(secretName1, secretName2, secretName3, secretName4);
   }
 
+  // secretsWithMatchingKeysAreReturned: search([k1], []) returns [secret1, secret2]
   @Test
-  void listAndMapSingleSecretWithMultipleTags() {
-    createSecret(false, true, false);
-
+  void listAndMapSecretsWithMatchingTagKeys() {
     final Collection<AbstractMap.SimpleEntry<String, String>> secretEntries =
         awsSecretsManagerExplicit.mapSecrets(
-            secretTags.keySet().stream().collect(Collectors.toList()),
-            secretTags.values().stream().collect(Collectors.toList()),
-            AbstractMap.SimpleEntry::new);
-
-    secretNames.forEach(secretName -> validateMappedSecret(secretEntries, secretName));
+            List.of("tagKey1"), Collections.emptyList(), AbstractMap.SimpleEntry::new);
+
+    final List<String> secretNames =
+        secretEntries.stream()
+            .map(entry -> String.valueOf(entry.getKey()))
+            .collect(Collectors.toList());
+    assertThat(secretNames)
```

Comment:
This doesn't prove that we actually fetched any secrets or applied the mapping function. Can we check that the results can the expected secrets as well?